main.go

```diff
@@ -532,149 +550,235 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
-		ch <- prometheus.MustNewConstMetric(e.lruHotPercent, prometheus.GaugeValue, parse(settings, "hot_lru_pct"))
-		ch <- prometheus.MustNewConstMetric(e.lruWarmPercent, prometheus.GaugeValue, parse(settings, "warm_lru_pct"))
-		ch <- prometheus.MustNewConstMetric(e.lruHotMaxAgeFactor, prometheus.GaugeValue, parse(settings, "hot_max_factor"))
-		ch <- prometheus.MustNewConstMetric(e.lruWarmMaxAgeFactor, prometheus.GaugeValue, parse(settings, "warm_max_factor"))
+		if err := e.parseAndNewMetric(ch, e.maxConnections, prometheus.GaugeValue, settings, "maxconns"); err != nil {
+			parseError = err
+		}
+
+		if v, ok := settings["lru_crawler"]; ok && v == "yes" {
+			err := firstError(
+				e.parseBoolAndNewMetric(ch, e.lruCrawlerEnabled, prometheus.GaugeValue, settings, "lru_crawler"),
+				e.parseAndNewMetric(ch, e.lruCrawlerSleep, prometheus.GaugeValue, settings, "lru_crawler_sleep"),
+				e.parseAndNewMetric(ch, e.lruCrawlerMaxItems, prometheus.GaugeValue, settings, "lru_crawler_tocrawl"),
+				e.parseBoolAndNewMetric(ch, e.lruMaintainerThread, prometheus.GaugeValue, settings, "lru_maintainer_thread"),
+				e.parseAndNewMetric(ch, e.lruHotPercent, prometheus.GaugeValue, settings, "hot_lru_pct"),
+				e.parseAndNewMetric(ch, e.lruWarmPercent, prometheus.GaugeValue, settings, "warm_lru_pct"),
+				e.parseAndNewMetric(ch, e.lruHotMaxAgeFactor, prometheus.GaugeValue, settings, "hot_max_factor"),
+				e.parseAndNewMetric(ch, e.lruWarmMaxAgeFactor, prometheus.GaugeValue, settings, "warm_max_factor"),
+			)
+			if err != nil {
+				parseError = err
+			}
+		}
+	}
+	return parseError
+}
+
+func (e *Exporter) parseAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, labelValues ...string) error {
+	return e.extractValueAndNewMetric(ch, desc, valueType, parse, stats, key, labelValues...)
+}
+
+func (e *Exporter) parseBoolAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, labelValues ...string) error {
+	return e.extractValueAndNewMetric(ch, desc, valueType, parseBool, stats, key, labelValues...)
+}
+
+func (e *Exporter) extractValueAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, f func(map[string]string, string) (float64, error), stats map[string]string, key string, labelValues ...string) error {
+	v, err := f(stats, key)
+	if err == errKeyNotFound {
+		return nil
+	}
+	if err != nil {
+		return err
 	}
+
+	ch <- prometheus.MustNewConstMetric(desc, valueType, v, labelValues...)
+	return nil
 }
 
-func parse(stats map[string]string, key string) float64 {
-	v, err := strconv.ParseFloat(stats[key], 64)
+func parse(stats map[string]string, key string) (float64, error) {
+	value, ok := stats[key]
+	if !ok {
+		log.Errorf("Key not found: %s", key)
```

Comment:
I don't think we should log unavailable keys as errors, as this will be very noisy for older memcached versions. I believe only issues which should not occur for any supported memcached version should be logged as errors, like connection errors or failures to parse a returned value.

I suggest to use the debug level to log unavilable keys.